packages/types/src/identity/awsCredentialIdentity.ts

```diff
@@ -24,6 +24,11 @@ export interface AwsCredentialIdentity extends Identity {
    * AWS credential scope for this set of credentials.
    */
   readonly credentialScope?: string;
+
+  /**
+   * AWS accountId that may be used for endpoint routing.
```

Comment:
After looking at this, maybe we should only say `AWS account ID` (similar to access key ID and secret access key), and not tie it to endpoint routing only.

```suggestion
   * AWS account ID
```

Reply:
yeah, i had left that in conservatively so that it's easy to know why we introduced this now. but i agree generally, let's do it as per your suggestion.